src/handlers.ts

```diff
@@ -129,3 +130,37 @@ export async function logout(ctx: IRouterContext): Promise<any> {
   ctx.session = null // tslint:disable-line: no-expression-statement
   return Object.assign(ctx.response, { status: 200, body: { loggedOut: true } })
 }
+
+export async function subscribe(ctx: IRouterContext): Promise<any> {
+  const email: string = fromBody(ctx, 'email', 'string')
+
+  const { MC_LIST_ID, MC_API_KEY, MC_SERVER_PREFIX } = process.env
+
+  if (!MC_API_KEY || !MC_API_KEY || !MC_SERVER_PREFIX) {
+    throw { status: 500, message: 'Failed to subscribe. It is us, not you. Please contact us to get help' }
+  }
+
+  // tslint:disable-next-line: no-expression-statement
+  mailchimp.setConfig({
+    apiKey: MC_API_KEY,
+    server: MC_SERVER_PREFIX,
+  })
+
+  try {
+    // "pending" status means a confirmation email will be
+    // sent before adding the email to the list
+    // tslint:disable-next-line: no-expression-statement
+    await mailchimp.lists.addListMember(MC_LIST_ID, {
+      email_address: email,
+      status: 'pending',
+    })
+  } catch (error) {
+    // tslint:disable-next-line: no-expression-statement
+    console.error({ error })
```

Comment:
Do we need to catch here at all? 500 errors will get logged if thrown and if there's already a fine error message attached this seems unnecessary.